### PR TITLE
Remove profile controllers

### PR DESCRIPTION
**Pull request by dunglas**
Remove profile controllers and the dep to API Platform. This is not the responsibility of this bundle (it should be handled by the user app directly).

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -55,4 +55,3 @@ install:
 script:
     - vendor/bin/phpunit
     - vendor/bin/behat
-    - vendor/bin/behat --profile=api_platform
```
Comment (vincentchalamon) [inline .travis.yml]: You forgot to update `behat.yml` file

**Comment by vincentchalamon** [comment]
As `api-platform` event is removed, can you update `AppKernel::registerContainerConfiguration` to remove dynamic load of `config_{env}.yml` file? Also don't forget to clean config files name (`routing_test.yml` => `routing.yml`)

**Comment by vincentchalamon** [comment]
`UserBundle\Entity\User` is now useless, can you remove it?

**Comment by vincentchalamon** [comment]
Please also update user documentation

**Comment by dunglas** [comment]
@vincentchalamon done (I also refactored the doc).

**Comment by dunglas** [comment]
I've refactored the whole test app to use a microkernel instead (similar to what I've done in DunglasActionBundle: https://github.com/dunglas/DunglasActionBundle/blob/master/Tests/Fixtures/TestKernel.php).

File: .travis.yml (hunk 1 above)
Comment (dunglas) [inline .travis.yml]: done

**Comment by vincentchalamon** [comment]
Just remove the `default.suites.default.paths` key in `behat.yml` file, after it'll be 👍

**Comment by dunglas** [comment]
Done.

**Comment by vincentchalamon** [comment]
Thanks @dunglas

**Pull request merged**